src/java.desktop/share/classes/javax/swing/SwingUtilities.java

```diff
@@ -1126,8 +1126,13 @@ private static String layoutCompoundLabelImpl(
             }
             v = (c != null) ? (View) c.getClientProperty("html") : null;
             if (v != null) {
-                textR.width = Math.min(availTextWidth,
-                                       (int) v.getPreferredSpan(View.X_AXIS));
+                if(availTextWidth < (int) v.getPreferredSpan(View.X_AXIS)){
```

Comment:
Can you attach "before" and "after" images into the bug report as part of the evaluation ?
Do this for Aqua and Metal too.

I presume this code only gets entered for labels with images ?
If this is all L&Fs then please manually test on mac. Windows and GTK L&Fs using SwingSet2 looking for problems as well as running all automated client tests using mach5.

We do not use syntax like "if(" and "else{".
It should always be "if (" and "else {"
Also no space after (int) for the cast. lines 1129 and 1134 .. 
And make sure lines are <=80 characters long which that one 1134 doesn't look to be.

Reply:
Images will be attached.

This code runs for html images as well as html text. I found that the extra padding issue exists for anything bound by the <html> tags. 

Syntax has been fixed locally and will appear in next commit.

Reply:
> @prrace Also no space after (int) for the cast. lines 1129 and 1134 ..

I'd like to clarify what style is followed by the client code.

[Code Conventions for the Java](https://www.oracle.com/java/technologies/javase/codeconventions-contents.html) published in 1999 in its section for white-space [state](https://www.oracle.com/java/technologies/javase/codeconventions-whitespace.html#498):

_“Casts should be followed by a blank space.”_

A more recent [draft for Java Style](http://cr.openjdk.java.net/~alundblad/styleguide/index-v6.html) that is commonly referred to during code reviews also says about using the space after the cast in its [Horizontal Whitespace section](http://cr.openjdk.java.net/~alundblad/styleguide/index-v6.html#toc-whitespace):

_“A single space should be used…  
After the closing parenthesis of a cast.”_

As far as I can see, neither style — *no space* after the cast vs *a space* after the cast — is followed consistently. Let's agree on one option and follow it then.

Shall I bring it up for a discussion on the client-libs mailing list?